src/rez/tests/test_solver.py

```diff
@@ -256,6 +256,21 @@ def test_12_missing_variant_requires(self):
         config.override("error_on_missing_variant_requires", False)
         self._solve(["missing_variant_requires"], ["nada[]", "missing_variant_requires-1[1]"])
 
+    def test_13_resolve_weakly_reference_requires(self):
```

Comment:
Can we add a comment to say something more clearly describe what's being tested please?